### ISSUE TITLE
compatibility policy / support timeframe

### ISSUE THREAD

**Issue by ThomasWaldmann** [jborg/attic#215]
A question that came up considering new ciphers/macs/compression/etc.:

How long shall attic support backups created by older attic versions?

Forever? That would either mean code gets more and more complex OR that development is blocked from supporting new stuff.

I see the need to be able to read old backups, but "forever" seems to be too much / too harmful.

In any case, defining this policy would be useful, currently it is (AFAIK) undefined.

**Comment by dnnr** [comment]
If it's going to be anything other than "forever", there should at least be a tool or command available for converting existing repositories to a new scheme.

**Comment by ThomasWaldmann** [comment]
@dnnr good idea. as there would be some overlap (old and new features supported in version x), conversion could be done using that version without needing too much code.

**Comment by adept** [comment]
In any sort of realistic scenario that would rather be "versions x to y"
(it is unreasonable to expect all users to upgrade sequentially, and
upgrade within some well-defined time window).

So overall I think it is inevitable that code adding support for new
features would have to contain lots of backwards-compatibility measures,
and it is expected that they would be there for a while.

I think a while ago there was a discussion on this list about potential
change of repository format, and some of these issues were rehashed there.
Sadly, I can't find the link right now :(

On Wed, Mar 4, 2015 at 2:50 PM, TW <notifications@github.com> wrote:

> @dnnr <https://github.com/dnnr> good idea. as there would be some overlap
> (old and new features supported in version x), conversion could be done
> using that version without needing too much code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jborg/attic/issues/215#issuecomment-77170653>.
>



-- 
Dmitry Astapov

**Comment by anarcat** [comment]
the only thing i've seen about this is add59e4ab3a38dcd78746bd865c52d4322c95e5f so far, which does point at potential, backwards-incompatible, archive changes in the future.

**Comment by jborg** [comment]
I think "forever" should be the goal, and something that most people would expect from a backup program. GNU tar for example has no problem restoring a dump from the 70's.

That said, it might not always be feasible and some kind of conversion program might be good enough in some cases. But I still think we should aim at "forever".

**Comment by anarcat** [comment]
thanks @jborg - that's great news! and i totally agree with the tar philosophy... however, the 70's tar cannot read today's extended formats... it may still extract files though, which is pretty amazing.
